File: src/test/java/com/kobylynskyi/graphql/codegen/scala/compile-pass/EventByIdQueryResponse.scala
import com.kobylynskyi.graphql.codegen.model.graphql.GraphQLResult
import java.util.{ Map => JMap }

/**
 * Single event by ID.
 */
class EventByIdQueryResponse extends GraphQLResult[JMap[String, Event]] {

    /**
     * Single event by ID.
     */
    def eventById(): Event = {
        val data: JMap[String, Event] = getData
        if (data != null) data.get(EventByIdQueryResponse.OPERATION_NAME) else null
    }

}

object EventByIdQueryResponse {

    private final val OPERATION_NAME: String = "eventById"

}

File: src/test/java/com/kobylynskyi/graphql/codegen/kotlin/compile-pass/EventByIdQueryResponse.kt
import com.kobylynskyi.graphql.codegen.model.graphql.GraphQLResult

/**
 * Single event by ID.
 */
open class EventByIdQueryResponse : GraphQLResult<MutableMap<String, Event>>() {

    companion object {
        const val OPERATION_NAME: String = "eventById"
    }

    /**
     * Single event by ID.
     */
    fun eventById(): Event {
        val data: MutableMap<String, Event> = super.getData()
        return data.getValue(OPERATION_NAME)
    }
}
